Sources/MongoSwift/ReadWriteConcern.swift

```diff
@@ -114,3 +113,166 @@ extension ReadConcern: Equatable {
         return lhs.level == rhs.level
     }
 }
+
+/// A class to represent a MongoDB write concern.
+public class WriteConcern: Codable {
+
+    /// A pointer to a mongoc_write_concern_t
+    internal var _writeConcern: OpaquePointer?
+
+    /// An option to request acknowledgement that the write operation has propagated to specified mongod instances.
+    public enum W: Codable, Equatable {
+        /// Specifies the number of nodes that should acknowledge the write. MUST be greater than or equal to 0.
+        case number(Int32)
+        /// Indicates a tag for nodes that should acknowledge the write. 
+        case tag(String)
+        /// Specifies that a majoirty of nodes should acknowledge the write.
+        case majority
+
+        public init(from decoder: Decoder) throws {
+            let container = try decoder.singleValueContainer()
+            if let wTag = try? container.decode(String.self) {
+                self = .tag(wTag)
+            } else {
+                let wNumber = try container.decode(Int32.self)
+                if wNumber == MONGOC_WRITE_CONCERN_W_MAJORITY {
+                    self = .majority
+                } else {
+                    self = .number(wNumber)
+                }
+            }
+        }
+
+        public func encode(to encoder: Encoder) throws {
+            var container = encoder.singleValueContainer()
+            switch self {
+            case let .number(wNumber):
+                try container.encode(wNumber)
+            case let .tag(wTag):
+                try container.encode(wTag)
+            case .majority:
+                try container.encode("majority")
+            }
+        }
+
+        public static func == (lhs: W, rhs: W) -> Bool {
+            switch (lhs, rhs) {
+            case let (.number(lNum), .number(rNum)):
+                return lNum == rNum
+            case let (.tag(lTag), .tag(rTag)):
+                return lTag == rTag
+            case (.majority, .majority):
+                return true
+            default:
+                return false
+            }
+        }
+    }
+
+    /// Indicates the `W` value for this `WriteConcern`.
+    public var w: W {
+        if let wTag = mongoc_write_concern_get_wtag(self._writeConcern) {
```

Comment:
Would it make sense to implement logic similar to `MONGOC_WRITE_CONCERN_W_MAJORITY` and instead check whether `mongoc_write_concern_get_w()` returns `MONGOC_WRITE_CONCERN_W_TAG`?